### PR TITLE
convert generic tasks to pretrained models

### LINKED ISSUE #972: Convert non-classification snapshots to "pretrained models"
Looks like https://github.com/NVIDIA/DIGITS/pull/932 doesn't let me convert other types of models like those for object detection or semantic segmentation.

### PR DESCRIPTION
**Pull request by Lucaszw**
Close #972

### CODE REVIEW

**Comment by lukeyeager** [comment]
I'm hitting this again (also here: https://github.com/NVIDIA/DIGITS/pull/937#issuecomment-239536218):
```
  File "/home/lyeager/digits/digits/model/views.py", line 216, in to_pretrained
    job.dir() + "/" + task.model_file ,
TypeError: cannot concatenate 'str' and 'NoneType' objects
```
Is this fixed in #937? Which do you want me to merge first?

**Comment by Lucaszw** [comment]
Hi Luke, yes it probably would be best to merge #937 . Its also includes a fix so that all models have model_file vs just new ones.

File: digits/model/views.py [hunk 1]
```diff
@@ -202,13 +202,22 @@ def to_pretrained(job_id):
     snapshot_filename = None
     snapshot_filename = task.get_snapshot(epoch)
 
+    # Set defaults:
+    labels_path = None
+    resize_mode = None
+
+    if "labels file" in info:
+        labels_path = task.dataset.dir() + "/" + info["labels file"]
```
Comment (gheinrich) [inline digits/model/views.py]: as @TimZaman commented `os.path.join` is preferred.

**Comment by gheinrich** [comment]
Isn't it easier to just remove the notions of `labels_path`, `resize_mode`, etc. from pre-trained jobs? I don't see where they are used. Besides, those would be properties of the dataset, not the model. Admittedly they are part of the model snapshot because they're needed for inference. However if we can't perform inference on a pre-trained model I think those fields are just getting in the way of simplicity.

**Comment by Lucaszw** [comment]
@gheinrich The labels_path is used for writing the deploy file for caffe (although only for getting the number of outputs, so there are definitely better ways to do that). Although for now resize_mode isn't used (I kept it in case we decided to add test images for inference on pre-trained models). Although it might be best left for another PR as its currently expected for classification jobs as well.

**Comment by gheinrich** [comment]
Ah and the deploy file is just used when visualizing layers, correct?

**Comment by Lucaszw** [comment]
@gheinrich Yeah for visualizing, and for gradient ascent (#968)

**Pull request merged**